kolibri/core/views.py

```diff
@@ -3,10 +3,13 @@
 from django.core.urlresolvers import translate_url
 from django.utils.http import is_safe_url
 from django.utils.translation import LANGUAGE_SESSION_KEY, check_for_language
+from django.views.decorators.csrf import csrf_exempt, ensure_csrf_cookie
 from django.views.i18n import LANGUAGE_QUERY_PARAMETER
 
 
 # Modified from django.views.i18n
+@csrf_exempt
```

Comment:
This does fix the issue, but I'm not convinced it's the right approach.

The problem seems to be that no CSRF cookie is set when a user is not logged in! This seems odd.

Could you see what we are missing that is allowing this to happen instead?

Reply:
@benjaoming any suggestions for why this cookie might not be set for a non-logged in user?

Reply:
Okay @rtibbles, Thanks for the explanation.

Reply:
@rtibbles you have to set it in the POST data. It might be that the cookie isn't even set before the request is made.

Specifically read this:
https://docs.djangoproject.com/en/dev/ref/csrf/#page-uses-ajax-without-any-html-form

..and otherwise, read the whole thing @mrpau-richard - it's good to understand how CSRF works before fixing issues related to it. Removing it is almost never the answer, there's actually also a section in the docs about that.

Reply:
Right, seems like we need to set ensure_csrf_cookie on every view that renders a template? Is there a way to turn that on by default?

Reply:
Something like this might be the right solution? https://stackoverflow.com/questions/32231607/django-forcing-csrf-token-on-all-responses an additional Django middleware that ensures that the CSRF cookie is set on every response.

Reply:
@benjaoming to note, we set the CSRF token on every AJAX request to the server, the issue here is that the CSRF cookie is not present in the browser when a user is not logged in.

Reply:
Adding {% csrf_token %} to our base template should cover it.
https://docs.djangoproject.com/en/1.11/ref/csrf/#acquiring-the-token-if-csrf-use-sessions-is-true

Reply:
(That will trigger the generation of a session and set the cookie)

Reply:
@rtibbles & @benjaoming adding `{% csrf_token %}` at the base template will solve the issue. Thanks to you @jamalex